minidump-processor/src/processor.rs

```diff
@@ -1144,44 +1199,122 @@ fn try_detect_null_pointer_in_disguise(memory_accesses: Option<&[MemoryAccess]>)
     None
 }
 
-/// Bit-flip detection.
-mod bitflip {
+pub mod memory_operation {
     use super::*;
-    use crate::PossibleBitFlip;
 
     /// The memory operation occurring when a crash occurred.
-    #[derive(Default, PartialEq, Eq, Clone, Copy)]
+    #[derive(Clone, Copy, Debug, Eq, PartialEq)]
     pub enum MemoryOperation {
-        #[default]
-        Unknown,
+        NoOperation,
         Read,
         Write,
         Execute,
+        UnknownOperation,
+    }
+
+    impl std::fmt::Display for MemoryOperation {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            write!(
```

Comment:
FYI if you're just writing string constants you can [use `f.write_str` directly](https://doc.rust-lang.org/nightly/std/fmt/struct.Formatter.html#method.write_str).